service/src/test/java/bio/terra/cbas/util/TestBearerTokenFilter.java

```diff
@@ -0,0 +1,64 @@
+package bio.terra.cbas.util;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletRequestWrapper;
+import javax.servlet.ServletResponse;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpHeaders;
+import org.springframework.mock.web.MockFilterChain;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+class TestBearerTokenFilter {
+  private static final BearerTokenFilter filter = new BearerTokenFilter();
+
+  private ServletResponse response;
+  private FilterChain chain;
+
+  @BeforeEach
+  void initMocks() {
+    RequestContextHolder.resetRequestAttributes();
+    response = new MockHttpServletResponse();
+    chain = new MockFilterChain();
+  }
+
+  @Test
+  void testRequestWithAuthHeader() throws ServletException, IOException {
+    String tokenValue = "foo-token";
+    MockHttpServletRequest request = new MockHttpServletRequest();
+    request.addHeader(HttpHeaders.AUTHORIZATION, String.format("Bearer %s", tokenValue));
```

Comment:
Since you have the conditional in the code, you might as well test a case where there's no `Bearer` in the value to remove